tests/test_parser.py

```diff
@@ -257,6 +259,98 @@ def test_convert_to_date_cast_result_to_date(self, mocked_locale,
                          expected_output)
 
 
+class Test_bs_safeget(unittest.TestCase):
+    def test_bs_safeget_return_expected_text(self):
```

Comment:
the test is wrong. You are testing whole functionality here. You need to test single step only. Naming also states that it returns expected text, but looking to the code it returns the first element of a array

Reply:
renamed this test to `test_bs_safeget_return_expected_text_with_single_selector`
and added test `test_bs_safeget_return_expected_text_with_many_selectors`